src/hooks/useWalletSupportsChain/useWalletSupportsChain.ts

```diff
@@ -63,5 +67,17 @@ export const walletSupportsChain: UseWalletSupportsChain = ({ chainId, wallet })
   }
 }
 
+export const useWalletSupportsBuyAsset = () => {
+  const buyAssetChainId = useSwapperStore(selectBuyAsset).chainId
+  const wallet = useWallet().state.wallet
+  return walletSupportsChain({ chainId: buyAssetChainId, wallet })
+}
+
+export const useWalletSupportsSellAsset = () => {
```

Comment:
i'd be inclined to make this hook take an asset as a parameter rather than have 2 near duplicates

Reply:
Absolutely, `useWalletSupportsAsset` or `useWalletSupportsChainId` would generalize better.

Reply:
Removed helpers in https://github.com/shapeshift/web/pull/4479/commits/5c61fcb2a885f96d99a3fb2122523a55c79412fb